File: DemoServer/Demos/Advanced/FullTextSearch.cs
using System;
using System.Linq;
using DemoServer.Controllers;
using DemoServer.Entities;
using DemoServer.Helpers;
using DemoServer.Indexes;
using Microsoft.AspNetCore.Mvc;
using Raven.Client.Documents;
using Raven.Client.Documents.Session;

namespace DemoServer.Demos.Advanced
{
    public partial class AdvancedController : BaseController
    {
        [HttpGet]
        [Route("/advanced/fullTextSearch")]
        [Demo("Full Text Search", DemoOutputType.Flatten, demoOrder: 200)]
        public object FullTextSearch(string searchTerm = "Jazz")
        {
            using (var session = DocumentStoreHolder.MediaStore.OpenSession())
            {
                QueryStatistics stats;
                var results = session.Query<LastFmAnalyzed.Result, LastFmAnalyzed>()
                    .Statistics(out stats)
                    .Search(x => x.Query, searchTerm)
                    .OfType<LastFm>()
                    .Select(x => new LastFm
                    {
                        Artist = x.Artist,
                        Tags = x.Tags,
                        TimeStamp = x.TimeStamp,
                        Title = x.Title,
                        TrackId = x.TrackId
                    })
                    .Take(20)
                    .ToList();

                ServerTime = TimeSpan.FromMilliseconds(stats.DurationInMs);

                return results;
            }
        }
    }
}

File: DemoServer/JavaDemo/src/main/java/net/ravendb/demo/advanced/FullTextSearch.java
package net.ravendb.demo.advanced;

import net.ravendb.client.documents.DocumentStore;
import net.ravendb.client.documents.session.IDocumentSession;
import net.ravendb.client.documents.session.QueryStatistics;
import net.ravendb.client.primitives.Reference;
import net.ravendb.demo.DocumentStoreHolder;
import net.ravendb.demo.entities.LastFm;
import net.ravendb.demo.indexes.LastFmAnalyzed;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;

@Controller
public class FullTextSearch {

    @GetMapping("/advanced/fullTextSearch")
    public List<LastFm> fullTextSearch(
            @RequestParam(value = "searchTerm", defaultValue = "Jazz") String searchTerm) {

        try (IDocumentSession session = DocumentStoreHolder.getMediaStore().openSession()) {
            Reference<QueryStatistics> statsRef = new Reference<>();
            List<LastFm> results = session.query(LastFmAnalyzed.Result.class, LastFmAnalyzed.class)
                    .statistics(statsRef)
                    .search("Query", searchTerm)
                    .ofType(LastFm.class)
                    .selectFields(LastFm.class, "Artist", "Tags", "TimeStamp", "Title", "TrackId")
                    .take(20)
                    .toList();

            long serverTime = statsRef.value.getDurationInMs();

            return results;
        }
    }
}
